tests/test_all_rules.py

```diff
@@ -640,6 +640,37 @@ def test_deprecated_rules_modified(self):
         if result:
             self.fail(f"Deprecated rules {result} has been modified")
 
+    def test_rule_change_has_updated_date(self):
+        """Test to ensure modified rules have updated_date field updated."""
+
+        rules_path = get_path("rules")
+        rules_bbr_path = get_path("rules_building_block")
+
+        # Use git diff to check if the file(s) has been modified in rules/ rules_build_block/ directory
+        # For now this checks even rules/_deprecated any modification there will fail
+        # the test case "test_deprecated_rules_modified", which means an ignore directory
+        # is not required as there is a specific test for deprecated rules
+
+        detection_rules_git = make_git()
+        result = detection_rules_git("diff", "--diff-filter=M", "origin/main", "--name-only",
+                                     rules_path, rules_bbr_path)
+
+        # If the output is not empty, then file(s) have changed in the directory(s)
+        if result:
+            modified_rules = result.splitlines()
+            failed_rules = []
+            for modified_rule_path in modified_rules:
+                diff_output = detection_rules_git('diff', 'origin/main', modified_rule_path)
```

Comment:
- Line 655 performs a diff on directory to fetch names of all the files modified with `--name-only`. 
- Then we iterate over each file to take a diff of current state to origin/main, where all changes would be listed and it checks for updated_date to be present. If we do a single diff and then try to see what rules have changed and which of them have updated_date, its going to make the workflow unnecessarly complex. 
- If you imagine a dev workflow, where the first commit the date is updated and Unit test run, this test case passes. 
```
❯ git diff --diff-filter=M origin/main --name-only rules/ rules_building_block/                      
rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
(.venv) 
detection-rules on  tarde_118 [$!?] is 📦 v0.1.0 via 🐍 v3.12.3 (.venv) on ☁️  shashank.suryanarayana@elastic.co 
 git diff origin/main rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
diff --git a/rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml b/rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
index ff88b1d33..3607d488e 100644
--- a/rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
+++ b/rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
@@ -3,13 +3,13 @@ bypass_bbr_timing = true
 creation_date = "2022/11/01"
 integration = ["endpoint", "windows"]
 maturity = "production"
-updated_date = "2024/05/21"
+updated_date = "2024/06/21"
 
 [rule]
 author = ["Elastic"]
 building_block_type = "default"
 description = """
-Identifies the execution of discovery commands to enumerate system information, files, and folders using the Windows
+Identifies the execution of discovery commands to enumerate system information, files, and folders using the Windows modified
 Command Shell.
 """
 from = "now-9m"
```
- Then in the next commit u modify something more but dont touch updated_date [ which is what most devs do] the next diff since its comapred with origin/main and not with the previous commit produces the output where u find updated_date
```
 git diff origin/main rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
diff --git a/rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml b/rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
index ff88b1d33..850fcbc76 100644
--- a/rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
+++ b/rules_building_block/discovery_files_dir_systeminfo_via_cmd.toml
@@ -3,13 +3,13 @@ bypass_bbr_timing = true
 creation_date = "2022/11/01"
 integration = ["endpoint", "windows"]
 maturity = "production"
-updated_date = "2024/05/21"
+updated_date = "2024/06/21"
 
 [rule]
 author = ["Elastic"]
-building_block_type = "default"
+building_block_type = "default_modified"
 description = """
-Identifies the execution of discovery commands to enumerate system information, files, and folders using the Windows
+Identifies the execution of discovery commands to enumerate system information, files, and folders using the Windows modified
 Command Shell.
 """
 from = "now-9m"
(.venv) 
detection-rules on  tarde_118 [$!?] is 📦 v0.1.0 via 🐍 v3.12.3 (.venv) on ☁️  shashank.suryanarayana@elastic.co 
❯ ```
- So even if many commits to rule changes updated_date only once in any of the commit, the test case can effectively catch those and provide results.